client/purplecaffeine/backend.py

```diff
@@ -1,14 +1,16 @@
 """Backend."""
+import json
 
 
 class BaseBackend:
     """Base backend class."""
 
-    def save_trial(self, trial):
+    def save_trial(self, name: str, trial_json: json):
```

Comment:
Let's keep type `Trial` as later on we can change serialization from json to more performant binary format :)